pyccel/ast/core.py

```diff
@@ -4298,6 +4375,62 @@ def execute(self):
 
 # TODO: Should Declare have an optional init value for each var?
 
+class FuncAddressDeclare(Basic):
+
+    def __new__(
+        cls,
+        variable,
+        intent=None,
+        value=None,
+        static=False,
+        ):
+
+        if not isinstance(variable, FunctionAddress):
+            raise TypeError('variable must be of type FunctionAddress, given {0}'.format(variable))
+
+        if intent:
+            if not intent in ['in', 'out', 'inout']:
+                raise ValueError("intent must be one among {'in', 'out', 'inout'}")
+
+        if not isinstance(static, bool):
+            raise TypeError('Expecting a boolean for static attribute')
+
+        return Basic.__new__(
+            cls,
+            variable,
+            intent,
+            value,
+            static,
+            )
```

Comment:
For new classes we would like to avoid using `__new__` as the `_args` tuple is hard to maintain. Please use `__init__` (see `Variable` class for how to handle `__new__` in this case)